secp256kfun/src/point.rs

```diff
@@ -114,6 +114,174 @@ impl Point<Normal, Public, NonZero> {
+    /// - Hash-to-curve typically represents a small fraction of total computation
+    /// - The maximum slowdown is bounded and relatively modest
+    /// - Creating adversarial inputs requires significant computational resources
+    ///
+    /// **For most use cases, prefer [`hash_to_curve`]** which is simpler and faster.
+    /// Only use this method if you have specific DoS concerns and hash-to-curve
+    /// represents a significant portion of your protocol's computation.
+    ///
+    /// **HAZMAT WARNING**: It is this author's opinion that [RFC 9380] is overwrought for
+    /// secp256k1. While this implementation passes test vectors from the
+    /// [`k256`](https://github.com/RustCrypto/elliptic-curves/tree/master/k256) crate (see their [test vectors](https://github.com/RustCrypto/elliptic-curves/blob/3381a99b6412ef9fa556e32a834e401d569007e3/k256/src/arithmetic/hash2curve.rs#L296)),
+    /// the complexity of the SSWU algorithm makes me hesitant to recommend its use.
+    /// The simpler try-and-increment method in [`hash_to_curve`] is preferred.
+    ///
+    /// # Parameters
+    /// - `msg`: The message to hash
+    /// - `dst`: Domain separation tag (DST), should be unique per application
+    ///
+    /// # Example
+    /// ```
+    /// # use secp256kfun::{Point, hash};
+    /// # use sha2::Sha256;
+    /// let point = Point::hash_to_curve_sswu::<Sha256>(b"hello world", b"myapp-v1");
+    /// ```
+    ///
+    /// [`hash_to_curve`]: Self::hash_to_curve
+    /// [`hash_to_curve_rfc9381_tai`]: Self::hash_to_curve_rfc9381_tai
+    /// [RFC 9380]: https://datatracker.ietf.org/doc/html/rfc9380
+    pub fn hash_to_curve_sswu<H>(msg: &[u8], dst: &[u8]) -> Point<NonNormal, Public, NonZero>
+    where
+        H: crate::hash::Hash32 + crate::digest::crypto_common::BlockSizeUser,
+    {
+        let backend_point = backend::Point::hash_to_curve::<H>(msg, dst);
+        Point::from_inner(backend_point, NonNormal)
+    }
+
+    /// Hash to curve using try-and-increment method
+    ///
+    /// This is a simple and efficient method to hash arbitrary byte strings to curve points
+    /// with uniform distribution. It works by hashing the input with an incrementing counter
+    /// until a valid curve point is found.
+    ///
+    /// **This is the recommended method for most applications.** While it has variable
+    /// runtime based on input (see [`hash_to_curve_sswu`] for details), this is rarely
+    /// a practical concern.
+    ///
+    /// ## Why not the [RFC 9381] try-and-increment?
+    ///
+    /// The VRF specification ([RFC 9381 §5.4.1.1](https://datatracker.ietf.org/doc/html/rfc9381#section-5.4.1.1))
+    /// includes a try-and-increment method (see [`hash_to_curve_rfc9381_tai`]) that always
+    /// uses a fixed y-coordinate parity (0x02). This results in a non-uniform distribution
+    /// that only includes points with even y-coordinates. Our implementation achieves
+    /// uniform distribution with a simple modification.
```

Comment:
Are there any consequences of only outputting even-y hash to curve points? If not, i'm thinking that using and recommending this RFC could be best and most widely supported? Even if the change is minor